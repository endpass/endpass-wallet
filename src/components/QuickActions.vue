<template>
  <div class="quick-actions">
    <div class="navbar is-fixed-bottom">
      <router-link
        active-class="is-active"
        class="navbar-item button"
        to="/"
        exact
      >
        <span
          class="icon is-small"
          v-html="require('@/img/home.svg')"
        />
        <span class="caption">Home</span>
      </router-link>
      <router-link
        :disabled="!wallet"
        :to="{name: 'SendPage'}"
        active-class="is-active"
        class="navbar-item button"
      >
        <span
          class="icon is-small"
          v-html="require('@/img/arrow-thick-left.svg')"
        />
        <span class="caption">Send</span>
      </router-link>
      <router-link
        :disabled="!address"
        :to="{name:
        'ReceivePage'}"
        class="navbar-item button"
        active-class="is-active"
      >
        <span
          class="icon is-small"
          v-html="require('@/img/arrow-thick-right.svg')"
        />
        <span class="caption">Recieve</span>
      </router-link>
      <router-link
        :disabled="!address"
        :to="{name: 'HistoryPage'}"
        class="navbar-item button"
        active-class="is-active"
      >
        <span
          class="icon is-small"
          v-html="require('@/img/clock.svg')"
        />
        <span class="caption">History</span>
      </router-link>
    </div>
  </div>
</template>

<script>
import { mapGetters, mapState } from 'vuex';

export default {
  name: 'quick-actions',
  data() {
    return {};
  },
  computed: {
    ...mapState({
      wallet: state => state.accounts.wallet,
      address: state =>
        state.accounts.address &&
        state.accounts.address.getChecksumAddressString(),
    }),
    ...mapGetters('accounts', ['isPublicAccount']),
  },
};
</script>

<style lang="scss">
.quick-actions {
  //flex: 1;

  .navbar {
    background: $white;
    display: flex;
    justify-content: space-between;
    align-items: stretch;

    .navbar-item {
      span {
        display: block;
        margin: 0 auto;
      }
    }

    .button {
      border: none;
      .icon {
        margin: 0 auto;
      }
    }
  }
}
</style>
