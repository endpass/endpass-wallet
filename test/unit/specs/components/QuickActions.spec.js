import { shallow, createLocalVue } from '@vue/test-utils';

import Vuex from 'vuex';
const localVue = createLocalVue();

localVue.use(Vuex);

import QuickActions from '@/components/QuickActions';

describe('QuickActions', () => {
  let wrapper;
  beforeEach(() => {
    let storeOptions = {
      modules: {
        accounts: {
          namespaced: true,
          state: {
            wallet: null,
            address: null,
          },
          getters: {
            isPublicAccount: jest.fn(),
          },
        },
      },
    };
    const store = new Vuex.Store(storeOptions);
    wrapper = shallow(QuickActions, {
      localVue,
      store,
    });
  });

  describe('render', () => {
    it('should be a Vue component', () => {
      expect(wrapper.isVueInstance()).toBeTruthy();
      expect(wrapper.name()).toBe('QuickActions');
    });

    it('should render initial state of the component', () => {
      expect(wrapper.element).toMatchSnapshot();
    });
  });
});
