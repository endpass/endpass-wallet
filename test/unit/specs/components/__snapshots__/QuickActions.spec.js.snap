// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`QuickActions render should render initial state of the component 1`] = `
<div
  class="quick-actions"
>
  <div
    class="navbar is-fixed-bottom"
  >
    <router-link
      active-class="is-active"
      class="navbar-item button"
      exact=""
      to="/"
    >
      <span
        class="icon is-small"
      >
        {}
      </span>
       
      <span
        class="caption"
      >
        Home
      </span>
    </router-link>
     
    <router-link
      active-class="is-active"
      class="navbar-item button"
      disabled="true"
      to="[object Object]"
    >
      <span
        class="icon is-small"
      >
        {}
      </span>
       
      <span
        class="caption"
      >
        Send
      </span>
    </router-link>
     
    <router-link
      active-class="is-active"
      class="navbar-item button"
      disabled="true"
      to="[object Object]"
    >
      <span
        class="icon is-small"
      >
        {}
      </span>
       
      <span
        class="caption"
      >
        Recieve
      </span>
    </router-link>
     
    <router-link
      active-class="is-active"
      class="navbar-item button"
      disabled="true"
      to="[object Object]"
    >
      <span
        class="icon is-small"
      >
        {}
      </span>
       
      <span
        class="caption"
      >
        History
      </span>
    </router-link>
  </div>
</div>
`;
